space_packet_parser/xtcedef.py

```diff
@@ -1581,33 +1593,69 @@ def __init__(self, size_in_bits: int, encoding: str = 'IEEE-754',
         if encoding not in self._supported_encodings:
             raise ValueError(f"Invalid encoding type {encoding} for float data. "
                              f"Must be one of {self._supported_encodings}.")
-        if encoding == 'MIL-1750A':
-            raise NotImplementedError("MIL-1750A encoded floats are not supported by this library yet.")
+        if encoding == 'MIL-1750A' and size_in_bits != 32:
+            raise ValueError("MIL-1750A encoded floats must be 32 bits, per the MIL-1750A spec. See "
+                             "https://www.xgc-tek.com/manuals/mil-std-1750a/c191.html#AEN324")
         if encoding == 'IEEE-754' and size_in_bits not in (16, 32, 64):
             raise ValueError(f"Invalid size_in_bits value for IEEE-754 FloatDataEncoding, {size_in_bits}. "
                              "Must be 16, 32, or 64.")
         super().__init__(size_in_bits, encoding=encoding, byte_order=byte_order,
                          default_calibrator=default_calibrator, context_calibrators=context_calibrators)
 
-        if self.byte_order == "leastSignificantByteFirst":
-            self._struct_format = "<"
+        if self.encoding == "MIL-1750A":
+            def _mil_parse_func(mil_bytes: bytes):
+                """Parsing function for MIL-1750A floats"""
+                # MIL 1750A floats are always 32 bit
+                # See: https://www.xgc-tek.com/manuals/mil-std-1750a/c191.html#AEN324
+                #
+                #  MSB                                         LSB MSB          LSB
+                # ------------------------------------------------------------------
+                # | S|                   Mantissa                 |    Exponent    |
+                # ------------------------------------------------------------------
+                #   0  1                                        23 24            31
+                bytes_as_int = int.from_bytes(mil_bytes, byteorder='big')
+                exponent = bytes_as_int & 0xFF  # last 8 bits
+                mantissa = (bytes_as_int >> 8) & 0xFFFFFFFF  # bits 0 through 23 (24 bits)
```

Comment:
Don't you only want this to be 6 Fs if you want the first 24 bits?

Reply:
Huh... Yeah. I guess this actually doesn't matter because when we bit shift `bytes_as_int`, it gets padded with zeros. I don't suppose a bitwise AND takes much time but I could just take it out. We do need the mask for the exponent.